### DISCUSSION TITLE
How should be specified a query/header parameter thats allow duplication

### DISCUSSION THREAD

**Discussion by joolfe** [OAI/OpenAPI-Specification#3031] [Q&A] [answered]
As long as I understand in HTTP is possible to send duplicated query and header parameters, how the server handle this is out of the scope of the specification but I have observed that the combination of "name" and location (expressed by field "in") in parameters in Open API should be unique so what is the correct way to specify that an API allow multiple query/headers, for example:

```
HTTP://myapi.com?id=123&id=345
```
Best regards

**Comment by flaksp** [comment, marked as answer]
"[4.8.12.3 Style Values](https://spec.openapis.org/oas/latest.html#style-values)" section covers how parameters may be represented with different data types (also check out table with examples in "[4.8.12.4 Style Examples](https://spec.openapis.org/oas/latest.html#style-examples)" section below). It is possible to represent array of query-parameters as you wish with `style: form`, `explode: true` and using schema of type `array` in `schema` keyword. 

Example:

```yaml
name: color
in: query
style: form
explode: true
schema:
  type: array
  uniqueItems: true
  items:
    enum: [blue, black, brown]
```
```
color=blue&color=black&color=brown
```

However, OpenAPI specification does not cover multiple headers with same name, so behaviour is undefined 😔 But it's nice idea to visit "[Big thoughts for 4.0](https://github.com/OAI/OpenAPI-Specification/discussions/2930)" discussion where we gather ideas for the next major update of the spec and suggest yours about headers with attached references to HTTP spec, existing issues and discussions if they exist.

**Reply by karenetheridge** [reply]
Multiple headers are supported too -- it would be `style: simple, explode: false` with `type: array`. See the examples at https://swagger.io/docs/specification/serialization/.

**Comment by handrews** [comment]
Looks like this was resolved, closing.
